src/scripts/main.js

```diff
@@ -1 +1,74 @@
 'use strict';
+
+const body = document.body;
+
+async function handlePromise(promise) {
+  try {
+    const message = await promise;
+
+    body.insertAdjacentHTML(
+      'beforeend',
+      `
+      <div data-qa="notification" class="success">
+        ${message}
+      </div>
+    `,
+    );
+  } catch (errorMessage) {
+    body.insertAdjacentHTML(
+      'beforeend',
+      `
+      <div data-qa="notification" class="error">
+        ${errorMessage}
+      </div>
+    `,
+    );
+  }
+}
+
+const firstPromise = new Promise((resolve, reject) => {
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+
+  document.documentElement.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+});
+
+handlePromise(firstPromise);
+
+const secondPromise = new Promise((resolve) => {
+  document.documentElement.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+  });
+
+  document.documentElement.addEventListener('contextmenu', () => {
+    resolve('Second promise was resolved');
```

Comment:
The secondPromise resolves on both 'click' and 'contextmenu' events. This means the promise will resolve twice if both events occur, which is not ideal. Consider using a flag to ensure it resolves only once.